buildkitd/buildkitd.go

```diff
@@ -219,6 +219,7 @@ func Start(ctx context.Context, image string, settings Settings, reset bool) err
 
 	args = append(args,
 		"-e", fmt.Sprintf("CACHE_SIZE_MB=%d", settings.CacheSizeMb),
+		"-e", fmt.Sprintf("GIT_URL_INSTEAD_OF=%s", settings.GitURLInsteadOf),
```

Comment:
I *think* this no longer works in the updated buildkit, because we use `git config` and the buildkit git commands no longer use the global config?

Reply:
I put in a change to disable ssh configs (see https://github.com/moby/buildkit/blob/master/source/git/gitsource.go#L580 ); but I'm unsure of any change that prevents the git config from being read. I'll need to test to see if the `git config` commands we use still work or not.

Reply:
I just gave it a test with git, and it looks like it's being ignored by buildkit now. for instance I ran:

    GIT_URL_INSTEAD_OF="git@fitlab.com=git@gitlab.com:" ./build/linux/amd64/earth -VD gitlab.com/alexcb/hello+docker

which worked, and caused this to get fetched:

    internal | --> GIT CLONE git@gitlab.com:alexcb/hello.git

however when I run git directly in the buildkit docker container, I can see

    $ docker exec -ti earthly-buildkitd cat /root/.gitconfig
    [url "git@fitlab.com"]
    	insteadOf = git@gitlab.com:

and the clone fails with:

    $ docker exec -ti earthly-buildkitd git clone git@gitlab.com:alexcb/hello.git
    Cloning into 'hello'...
    fatal: 'git@fitlab.comalexcb/hello.git' does not appear to be a git repository
    fatal: Could not read from remote repository.

Reply:
Perhaps a short-term solution is to not ignore the config for now in the earthly buildkit fork and deal with this later.

Reply:
I added in HOME into the buildkit exec, so git will now pick up the global config: https://github.com/earthly/buildkit/commit/c1d00f6ed86584948c4563beab90964d19119046